homeassistant/components/bang_olufsen/media_player.py

```diff
@@ -358,6 +387,82 @@ def _async_update_volume(self, data: VolumeState) -> None:
 
         self.async_write_ha_state()
 
+    @callback
+    async def _async_update_beolink(self) -> None:
+        """Update the current Beolink leader, listeners, peers and self."""
+
+        # Add Beolink listeners / leader
+
+        self._remote_leader = self._playback_metadata.remote_leader
+
+        # Create group members list
+        group_members = []
+
+        # If the device is a listener.
+        if self._remote_leader is not None:
+            # Add leader
+            group_members.append(
+                cast(str, self._get_entity_id_from_jid(self._remote_leader.jid))
+            )
+
+            # Add self
+            group_members.append(
+                cast(str, self._get_entity_id_from_jid(self._beolink_jid))
+            )
+
+        # If not listener, check if leader.
+        else:
+            beolink_listeners = await self._client.get_beolink_listeners()
+
+            # Check if the device is a leader.
+            if len(beolink_listeners) > 0:
+                # Add self
+                group_members.append(
+                    cast(str, self._get_entity_id_from_jid(self._beolink_jid))
+                )
+
+                # Get the friendly names for the listeners from the peers
+                group_members.extend(
+                    [
+                        cast(str, self._get_entity_id_from_jid(beolink_listener.jid))
+                        for beolink_listener in beolink_listeners
+                    ]
+                )
+
```

Comment:
The group member list should be the same. In the first case there will only be a maximum of 2 group members; whereas in the second case more than 2 are possible.  For example, if 3 speakers are joined together all three should have the same group members.  However, with the current code only the leader speaker will have 3, the other 2 will have themselves and the leader.

Reply:
Yes, that is unfortunately a limitation with the API. Only the leader has the complete overview of the session. If it is critical, the complete group can be formed from attributes from the other entities.

Reply:
Got it. Not ideal since data is wrong, but perhaps ok for now.